magnum_cluster_api/resources.py

```diff
@@ -1724,6 +1728,8 @@ def get_machine_deployment(
     cluster: magnum_objects.Cluster,
     node_group: magnum_objects.NodeGroup,
 ) -> objects.KubeadmControlPlane:
+    if cluster.stack_id is None:
+        return None
```

Comment:
same thing here, it will return nothing since nothing is matching and since the if statement is not equal to 1 then return None?

Reply:
yeap, i checked that `kubectl get deploy -l test1=null, test2=test2` doesn't return deployments which has only `test2=test2` label.